### DISCUSSION TITLE
下書き記事を一斉削除する方法の提供

### DISCUSSION THREAD

**Discussion by yarakos95** [increments/qiita-discussions#323] [要望・アイデア・不具合]
Qiita では、記事を書いている途中に自動的に下書きに保存されるようになっています。その影響で、`下書き記事` に下書きが大量にたまってしまっています。現在、この `下書き記事` に保存された下書きは 1 つずつ削除する必要があります。

「一斉削除」あるいは「選択した下書き記事の削除」を行う方法を提供してほしいです。

## 私の状況

私は、テストとして Qiita Markdown の表示がどのようになるか試すことがあります。このテストは記事として投稿するつもりはないため、下書きに保存されていても不要なものになります。
一方で、投稿する多くの記事はエディタを利用して編集するため、下書きに保存されずとも問題ありません。

このような使い方をしているため、下書き記事にどのような状態の記事が残っていても問題にはなりません。
しかし、不要な記事は消しておきたい性格のため、このような下書き記事も消したいです。1 つ 1 つを削除することはかなり骨が折れるため、複数記事の削除が同時にできるとありがたいです。

## 追記

@tsubolabo さまのご意見を見て、少し考え方を変えても良いかもしれないと思いました。私は、下書きをかなり軽視していましたが、これを重く見る人にとっては誤った操作を誘発する危険なボタンに思えてしまうようです。

今回、下書きにたまってしまう投稿するつもりのない記事は Qiita Markdown の表示を試すためでした。これであれば、デモページの提供であっても良いと思います。

commonmark\.js や markdown-it では次のようなデモページが提供されています。

- commonmark\.js のデモページ

    https://spec.commonmark.org/dingus/

- markdown-it のデモページ

    https://markdown-it.github.io/

このようなデモページであれば、新規投稿ページで試したり、テスト用の記事を作成する必要はありません。

**Comment by tsubolabo** [comment]
折角のご提案に水を差すようで恐縮ですが、これを実現できるようなボタンが装備されると、下書き機能を本来の目的に使っている会員にとっては、「自爆ボタン」に見えてしまいます。

　私は、現在２０タイトル以上の下書き記事をもっており、それらがいつか日の目を見ることを夢見て日々手入れをしています（なかなか進捗しませんが）。これが１回のボタン誤操作で消えてしまうような事態を考えると悲しくなります。

　私も、表示結果を試すだけのために投稿エディタを使うことがありますが、一回試すたびに一つの記事が下書き保存されてしまうような使い方はしていません。お試し専用の「テスト」というタイトルの一記事を用意し、その中にどんどん書き足していくことで表示結果を確認しています。「テスト」のサイズが大きくなり過ぎれば、要らない部分を領域選択して自由に一括消去できます。

**Reply by munierujp** [reply]
>これが１回のボタン誤操作で消えてしまうような事態を考えると悲しくなります。

現状でも、下書きを削除する場合は確認ダイアログは表示されています。

<img width="454" alt="スクリーンショット 2022-10-14 9 45 10" src="https://user-images.githubusercontent.com/20086673/195737190-adf6bdae-b36b-4b60-a3ba-7dbb64987518.png">

一斉削除機能が実装される場合にダイアログなし仕様になるとは考えづらく、１回のボタン誤操作で消えてしまうような事態にはならないかと思います。
もっとも、ダイアログの確認も誤ってOKしてしまうことはあると思うので、記事投稿時のように目立つ確認ダイアログを表示するなどの工夫はあるといいでしょう。

**Reply by tsubolabo** [reply]
>一斉削除機能が実装される場合にダイアログなし仕様になるとは考えづらく、１回のボタン誤操作で消えてしまうような事態にはならないかと思います。

　万一実装されるようなことになったとしても、削除時にはダイアログで再確認が出る仕様になるだろうというのは承知の上でのコメントです。

　要は、下書き保存件数が増えていくような使い方を避ければ良いだけの話なので、提案されているような機能は望まないということです。

**Reply by yarakos95** [reply]
@tsubolabo さま

確かに、下書きを本来の目的に使っている場合では、自爆を誘発するようなボタンに見えてしまいますね。私とは異なる使い方をしている人の視点が無かったため、そこまで考えが至っていませんでした。

これを受けて、本 Discussion の追記として、Qiita Markdown のデモページを提供する案を提示しました。これであれば、私にとって下書き記事を一斉に削除する操作は不要となり、自爆を誘発するボタンを設置する必要がなくなると考えました。

-----

これはまったくの余談ですが、

> お試し専用の「テスト」というタイトルの一記事を用意し、その中にどんどん書き足していくことで表示結果を確認しています。

確かに、テスト記事を投稿しておいて、それを逐次編集で表示結果を確認することが出来ます。しかし、新規投稿を作成した方が手数が少なく楽なため、このような事態になっていました。また、いちいちテスト記事のページに遷移する必要が無かった点も、新規記事ページでテストする一つの要因になっていたと考えています。

**Comment by tanaka-asami415** [comment]
@Yarakashi-Kikohshi 
ご意見ありがとうございます🙏
一括削除したいというご意見は他にもいただいているので、検討させていただきます！🙇
デモページについては、多数ご意見があれば検討したいと思います！

**Reply by tsubolabo** [reply]
反対意見があることもお忘れなく。
